File: base-forge-security-authorization-spring-boot-starter-autoconfigure/src/main/java/com/gewuyou/baseforge/security/authorization/autoconfigure/config/entity/SecurityAuthorizationProperties.java
package com.gewuyou.baseforge.security.authorization.autoconfigure.config.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * 安全授权属性
 *
 * @author gewuyou
 * @since 2025-01-08 14:42:33
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@ConfigurationProperties("base-forge.security.authorization")
public class SecurityAuthorizationProperties {
    /**
     * 排除路径
     */
    private String[] ignored = new String[0];
    /**
     * 进行鉴权的请求路径
     */
    private String requestUrl = "/auth/z";
}

File: base-forge-security-authorization-spring-boot-starter-autoconfigure/src/main/kotlin/com/gewuyou/baseforge/security/authorization/autoconfigure/config/entity/SecurityAuthorizationProperties.kt
package com.gewuyou.baseforge.security.authorization.autoconfigure.config.entity

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * 安全授权属性
 *
 * @author gewuyou
 * @since 2025-01-08 14:42:33
 */
@ConfigurationProperties("base-forge.security.authorization")
data class SecurityAuthorizationProperties(
    /**
     * 排除的URL
     */
    var ignoredUrls: Array<String?> = arrayOfNulls(0),

    /**
     * 请求URL(默认鉴权请求URL：/api)
     */
    var requestUrl: String = "/api"
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is SecurityAuthorizationProperties) return false
        if (!ignoredUrls.contentEquals(other.ignoredUrls)) return false
        if (requestUrl != other.requestUrl) return false
        return true
    }

    override fun hashCode(): Int {
        var result = ignoredUrls.contentHashCode()
        result = 31 * result + requestUrl.hashCode()
        return result
    }
}